subprojects/core/src/integTest/groovy/org/gradle/internal/classpath/BuildScriptClasspathIntegrationSpec.groovy

```diff
@@ -163,12 +163,11 @@ class BuildScriptClasspathIntegrationSpec extends AbstractIntegrationSpec implem
 
     @Requires(value = IntegTestPreconditions.NotEmbeddedExecutor, reason = "explicitly requests daemon")
     def "url connection caching is not disabled by default"() {
-
         given:
         buildFile << """
             task checkUrlConnectionCaching {
                 doLast {
-                    URL url = new URL("jar:file://valid_jar_url_syntax.jar!/")
+                    URL url = new URL("${buildFile.toURI().toASCIIString()}")
```

Comment:
Does this need to be any file URL to disable this?

Reply:
Technically, it can be _any URL_, period. As long as connection can be opened, this flag is a single static boolean. It's actually really weird because the connection itself is entirely unimportant, so this could've just been a static setter, but whoever wrote `URLConnection` decided not to do that.